pages/models/generations/Sources/VSC/EMTGridFormingVSC/index.md

```diff
@@ -0,0 +1,354 @@
+
+$$ \frac{i^{qd*}_c(s)}{\hat{v}^{qd}(s)} = \frac{1}{R_f + sL_f} $$
+
+</div>
+
+The closed-loop transfer function, considering $$G_c = K^{icl}_p + \frac{K^{icl}_i}{s}$$ is:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \frac{i^{qd}_s(s)}{i^{qd*}_s(s)} = \frac{G_c(s)}{(L_f + L_c)s + (R_f + R_c) + G_c(s)} $$
+
+</div>
+
+where $$i^{qd*}_c(s)$$ is the reference current for *q* or *d* axis at the converter side, and $$i^{qd}_c(s)$$ is the measured current. The controller gains can be selected as $$K^{icl}_p = \frac{L_f}{\tau_c}$$ and $$K^{icl}_i = \frac{R_f}{\tau_c}$$, where $$\tau_c$$ is the time constant of the current loop, such that the complete closed-loop transfer function is represented as the following first-order response:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \frac{i^{qd}_c(s)}{i^{qd*}_c(s)} = \frac{1}{\tau_c s + 1} $$
+</div>
+
+The resulting PI controller is the following:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ G_{c}(s) = K_p + \frac{K_i}{s} = \frac{L_f}{\tau_c} + \frac{R_f}{\tau_c s} $$
+
+</div>
+
+
+### Limitations of current
+
+The technical constraints of the VSC can be included in the controls using saturation blocks. Depending on the desired operation mode, the converter can be set to prioritize one of the current components. These operation modes are typically defined by the grid codes, although a possible implementation could be the following:
+
+* **Normal operation**: The converter will follow the $$i^q$$ component setpoint, prioritizing the active power, and then $$i^d$$ will be limited by the operational limits of the converter $$i^d_{max} = \sqrt{I_{max}^2 - \max{i^q, i^{q*}}} $$.
+* **Transient or fault operation**: The converter will now prioritize the $$i^d$$ component, which will follow its reference, and $$i^q_{max} = \sqrt{I_{max}^2 - \max{i^d, i^{d*}}^2} $$.
+
+
+### Modulation
+
+As in the [EMT Grid Following Model](../EMTGridFollowingVSC/), the converter voltage setpoint obtained can be considered directly the converter voltage, in what is called an Averaged Model, since it does not consider the switching process. However, a more complete EMT simulation will model the modulation performed to obtain the sinusoidal voltage from the DC source. The details can be consulted in Section 3.6 of the said model.
+
+
+## Open source implementations
+
+This model has been successfully implemented in :
+
+| Software      | URL | Language | Open-Source License | Last consulted date | Comments |
+| PSTess | [Link](https://github.com/sandialabs/snl-pstess/blob/master/pstess/gfma.m) | MATLAB | Copyright 2021 National Technology & Engineering Solutions of Sandia, LLC
+(NTESS) | 09/07/2024 | Droop control version, equivalent to REGFM_A1 from WECC accepted dynamic models|
+
```

Comment:
Values not populated correctly. Please self-review and update it

Reply:
corrected